sources/lib/api/gcp/bigquery/_external_table.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+
+"""Implemenents External Table functionality."""
+
+import gcp
+import _api
+import _csv_options
+import _job
+
+
+class ExternalTable(object):
+
+  def __init__(self, source, source_format='csv', csv_options=None,
```

Comment:
I think a static factory method might be better. FederatedTable.from_storage or something like that, so that when a different source is supported later, we can add another factory method.